src/main/java/com/faforever/client/login/LoginController.java

```diff
@@ -225,13 +238,17 @@ public void display() {
     usernameInput.setText(Strings.nullToEmpty(username));
     autoLoginCheckBox.setSelected(isAutoLogin);
 
-    if (loginPrefs.getAutoLogin() && !isNullOrEmpty(username) && !isNullOrEmpty(password)) {
-      login(username, password, true);
-    } else if (isNullOrEmpty(username)) {
-      usernameInput.requestFocus();
-    } else {
-      passwordInput.requestFocus();
-    }
+    initializeFuture.thenAccept(aVoid -> {
+      Assert.notNull(loginAllowed, "loginAllowed must be set here or you mixed up the futures.");
+
+      if (loginAllowed && loginPrefs.getAutoLogin() && !isNullOrEmpty(username) && !isNullOrEmpty(password)) {
+        login(username, password, true);
+      } else if (isNullOrEmpty(username)) {
+        usernameInput.requestFocus();
+      } else {
+        passwordInput.requestFocus();
+      }
+    });
```

Comment:
It looks like you should add an exception handler to this. Otherwise any exceptions, including the `Assert.notNull...` above will get swallowed right?

Reply:
Well the Assert is there to at least give some information in the logs if the application is in an invalid state. But there is no proper recovery for that. What would I do in the exceptionally case?

Reply:
The Assert is fine, but right now it wont get logged. If i remember this correctly, if a CompletableFuture throws an exception asynchronously, that will just go into the Threadpool Nirvana unless you always register an exception handler that prints any exceptions.

Reply:
Which is a horrible thing in general because there are porbably tons of code like this that don't register any exception handlers so all of these exceptions dont get logged. Unless there is some global handler which i don't think exists by default.

Reply:
Okay, I now log it as error and enable login instead

Reply:
Ok.
I think it was ok/better the way before (if you had just added an `exceptionally` lamda after the `thenRun`). Because if for example any of the other code in that lamda throws a different exception, it will still not get logged.
And in the future, we should have a global exception handler that makes sure that exceptions at least get logged. Then we could remove the `exceptionally` handlers everywhere in the coder where they just catch/log exceptions.